modules/nw-ovn-ipsec-north-south-enable.adoc

```diff
@@ -0,0 +1,125 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/configuring-ipsec-ovn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-ovn-ipsec-north-south-enable_{context}"]
+= Enabling IPsec encryption for external IPsec endpoints
+
+// This procedure requests installing Butane to prepare the machine config
+
+As a cluster administrator, you can enable IPsec encryption between the cluster and external IPsec endpoints. Because this procedure uses Butane to create machine configs, you must have the `butane` command installed.
+
+[NOTE]
+====
+After you apply the machine config, the Machine Config Operator reboots affected nodes in your cluster to rollout the new machine config.
+====
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in to the cluster with a user with `cluster-admin` privileges.
+* You have reduced the size of your cluster MTU by `46` bytes to allow for the overhead of the IPsec ESP header.
+* You have installed the `butane` utility.
+
+.Procedure
+
+As a cluster administrator, you can enable IPsec support external for IPsec endpoints.
+
+. Create an IPsec configuration file named `ipsec-endpoint-config.conf`. The configuration is consumed in the next step. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks#configuring-a-vpn-with-ipsec_securing-networks[Configuring a VPN with IPsec].
+
+. Create a machine config to apply the IPsec configuration to your cluster:
+
+.. To add the IPsec configuration, create Butane config files for control plane and worker nodes with the following contents:
++
+[source,terminal,subs="attributes+"]
+----
+$ for role in master worker; do
+  cat >> "99-ipsec-$\{role}-endpoint-config.bu" <<-EOF
+  variant: openshift
+  version: {product-version}.0
+  metadata:
+    name: 99-worker-ipsec-import-certs
+    labels:
+      machineconfiguration.openshift.io/role: $role
+  extensions:
+    - ipsec
+  systemd:
+    units:
+      - name: ipsec.service
```

Comment:
this is not good enough
as it will try to enable ipsec.service before the extension is installed 
https://issues.redhat.com/browse/OCPBUGS-17826

the solution we're currently using:
```
    systemd:
      units:
      - name: ipsecenabler.service
        enabled: true
        contents: |
         [Unit]
         Description=Enable ipsec service after os extension installation
         Before=kubelet.service

         [Service]
         Type=oneshot
         ExecStart=systemctl enable --now ipsec.service

         [Install]
         WantedBy=multi-user.target
```
this will deploy a systemd unit that will enable and start ipsec service.

Reply:
@yuvalk, so we need two different Butane files, one that does the _oneshot_ above, and one that handles the `cacert.p12` and `ipsec-endpoint-config.conf`?

Thanks!

Reply:
no, I think it can all be done in one mc
the workaround above is exactly to allow that.